pkg/probes/probes.go

```diff
@@ -82,3 +83,46 @@ func (probes *probesImpl) BeginHealthCheck() {
 		}
 	}()
 }
+
+// ProbeChecker convenient checker for standard k8s probes liveness/readiness
+type ProbeChecker interface {
+	Alive() (bool, error)
```

Comment:
Why not, error can be easily ignored in client code, and it's a common thing in Go to return an error alongside the main result.

Reply:
yes, you are right, but in your code you use the error only for log. We actually need only to know is probe server is alive. I think we can also log the error insede Alive(), Ready() methods.

Reply:
The current code allows the client code to decide whether to report an error. Otherwise there will be either no diagnostic or too much of it (since probes are checked on regular basis).

Reply:
> Otherwise there will be either no diagnostic or too much of it (since probes are checked on regular basis).
But you already log it always, take a look at https://github.com/networkservicemesh/networkservicemesh/pull/1641/files#diff-b305f1bf4a29da22b05096c2f948b079R111
Or I missed someting?

Reply:
- [x] do not return err from ProbeChecker methods